### DISCUSSION TITLE
`disabled:file:opacity-50` cannot be minified by LightningCSS

### DISCUSSION THREAD

**Discussion by spaceemotion** [tailwindlabs/tailwindcss#11758] [Help] [answered]
**What version of Tailwind CSS are you using?**

v3.3.3

**What build tool (or framework if it abstracts the build tool) are you using?**

vite v4.4.8

**What version of Node.js are you using?**

Node 20

**What browser are you using?**

Chrome

**What operating system are you using?**

Linux/Docker

**Reproduction URL**

- [LightningCSS Playground](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Atrue%2C%22nesting%22%3Atrue%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22%5Cn%5Cn.disabled%5C%5C%3Afile%5C%5C%3Aopacity-50%3A%3Afile-selector-button%3Adisabled%20%7B%5Cn%20%20%20%20opacity%3A%200.5%3B%5Cn%7D%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D)
- [Tailwind Playground](https://play.tailwindcss.com/ac9E7E79fC)

**Describe your issue**

While trying out the new LightingCSS minification mode in vite, I found that custom File Upload input styling breaks it, as it appears to not be supported by the standard (The only ticket similar to my issue I found was https://github.com/parcel-bundler/lightningcss/issues/305)? 

LightningCSS just cancels out with the CSS being invalid:

```
Invalid pseudo class after pseudo element, only user action pseudo classes (e.g. :hover, :active) are allowed
```

As it appears to not change the style at all, a more correct class to use would simply be `disabled:opacity-50`, but that affects the whole input, not just the upload button inside of it.

Is this an invalid modifier for CSS? Should I use an alternative instead?

Thanks in advance!

**Comment by adamwathan** [comment, marked as answer]
Hey! Just need to flip the modifiers:

https://play.tailwindcss.com/Ve1gruE4WI
